Dev file browser update

NolanT32 opened the pull request:
DO NOT MERGE

NolanT32 committed:
Dev Commit

Completed mime type task

NolanT32 committed:
Add "All supported types" default to file browser

This patch adds a new file type to the file dialog window that opens
when the user loads a local video file. The new file type is an "all
supported file formats" option which includes all supported mime types
on the user's system. This option is made to be the default option.

Before this patch, a warning message would appear when certain
directories were switched to using the macOS native file browser. To
rememedy this, the QT-specific file browser is used instead.

Testing Steps
  1. Run main.py to start the application.
  2. Click the load video option in the menu bar.
  3. Verify that an option to select "All supported formats" appears and
     is the default option.
  4. Verify that a video can be run with no errors.
  5. Verify that switching directories does not cause an exception to be
      thrown.

Type: Bug Fix
Type: New Feature

NolanT32 committed:
Modified File Browser

This patch changes how the file browser was implemented based on the reasons stated below.

In the acceptance criteria document Michael wanted the file browser to be able to support these 4-5 basic file types: *.mp4 *.avi *.wmv and *mov. He also wanted a default option to be able to select any of these 4 files as well as an option to singularly select each file.

Additionally, we ran into an error last sprint when trying to change to certain directories within the file browser. We wanted to make sure this exception was no longer thrown when switching directories. The file browser was changed to display a native qt file browser rather than using the specific machines file browser as the qt file browser only shows directories it can handle without exceptions.

Testing Steps
  1. Run main.py to start the application.
  2. Click the load video option in the menu bar.
  3. Verify that an option to select "all supported files" appears
  4. Verify that a video can be run with no errors.
  5. Verify that switching directories does not cause an exception to be
      thrown.

Type: Bug Fix

Briggs-Richardson commented on Controllers/controller.py:
```diff
@@ -77,7 +80,35 @@ def open_file_dialog(self):
+                continue
+
+        # Appends .wmv and a default option that allows all files to be selected.
+        filtered_types.append("WMV video (*.wmv)")
```
The default mime types should include every currently supported mime type and should not default to these static values.

Briggs-Richardson committed:
Add "All supported types" default to file browser

This patch adds a new file type to the file dialog window that opens
when the user loads a local video file. The new file type is an "all
supported file formats" option which includes all supported mime types
on the user's system. This option is made to be the default option.

Before this patch, a warning message would appear when certain
directories were switched to using the macOS native file browser. To
rememedy this, the QT-specific file browser is used instead.

Testing Steps
  1. Run main.py to start the application.
  2. Click the load video option in the menu bar.
  3. Verify that an option to select "All supported formats" appears and
     is the default option.
  4. Verify that a video can be run with no errors.
  5. Verify that switching directories does not cause an exception to be
      thrown.

Type: Bug Fix
Type: New Feature

NolanT32 committed:
Merge branch 'dev_file_browser' of https://github.com/MichaelKintscher/qualitative-coding-app into dev_file_browser
